pkg/apis/events/v1/cloudauditlogssource_types.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	gcpduckv1 "github.com/google/knative-gcp/pkg/apis/duck/v1"
+	kngcpduck "github.com/google/knative-gcp/pkg/duck/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudAuditLogsSource is a specification for a Cloud Audit Log event source.
+type CloudAuditLogsSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CloudAuditLogsSourceSpec   `json:"spec"`
+	Status CloudAuditLogsSourceStatus `json:"status"`
+}
+
+// Verify that CloudAuditLogsSource matches various duck types.
+var (
+	_ apis.Defaultable             = (*CloudAuditLogsSource)(nil)
+	_ apis.Validatable             = (*CloudAuditLogsSource)(nil)
+	_ kmeta.OwnerRefable           = (*CloudAuditLogsSource)(nil)
+	_ resourcesemantics.GenericCRD = (*CloudAuditLogsSource)(nil)
+	_ kngcpduck.Identifiable       = (*CloudAuditLogsSource)(nil)
+	_ kngcpduck.PubSubable         = (*CloudAuditLogsSource)(nil)
+	_ duckv1.KRShaped              = (*CloudAuditLogsSource)(nil)
+)
+
+const (
+	SinkReady apis.ConditionType = "SinkReady"
```

Comment:
let's add a comment:

// SinkReady has status True when CloudAuditLogsSource sink is ready.